### PR TITLE
Issue #320 : Add capacity constraint to `RuntimeConfig`

### PR DESCRIPTION
**Pull request by rishabhmonga**
Adding ResourceUnit validation

### CODE REVIEW

**Comment by rishabhmonga** [comment]
This PR provides a validation for `ResourceUnit` so as to disallow changes if the resource has `MemoryUnit` as unit of measure and not allow changes to `ResourceUnit` for a resource. Also to be noted, if the `ResourcePools` instance being provided to the `updateResourcePools()` does not contain a resource present in the configuration, the existing configuration for that resource will be retained.

File: core/src/main/java/org/ehcache/util/ResourcePoolMerger.java [hunk 1]
```diff
@@ -40,6 +41,9 @@ public ResourcePools validateAndMerge(ResourcePools existing, ResourcePools toBe
           existing.getPoolForResource(currentResourceType).isPersistent()) {
         throw new IllegalArgumentException("Persistence configuration cannot be updated");
       }
+      if(toBeUpdated.getPoolForResource(currentResourceType).getUnit() instanceof MemoryUnit) {
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/util/ResourcePoolMerger.java]: While this test makes sense as long as #40 is not implemented, I believe we can have one expressed in the same way the persistent check is made - disallow changing the resource unit altogether.

File: core/src/test/java/org/ehcache/util/ResourcePoolsMergerTest.java [hunk 2]
```diff
@@ -76,4 +73,24 @@ public void testUpdatingHeapSizeToLargerThanDisk() {
...
   }
+  
+  @Test
+  public void testUpdateResourceUnit() {
+    ResourcePoolsBuilder existingPoolsBuilder = ResourcePoolsBuilder.newResourcePoolsBuilder();
+    existingPoolsBuilder = existingPoolsBuilder.heap(20L, EntryUnit.ENTRIES).disk(500, MemoryUnit.MB);
+    ResourcePools existing = existingPoolsBuilder.build();
+
+    ResourcePoolsBuilder toBeUpdatedPoolsBuilder = ResourcePoolsBuilder.newResourcePoolsBuilder();
+    toBeUpdatedPoolsBuilder = toBeUpdatedPoolsBuilder.disk(2, MemoryUnit.GB);
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/util/ResourcePoolsMergerTest.java]: This test fails today but should pass in the future once ARC is implemented.
So I would replace it with a test checking that we do not try to switch a count based resource to a memory based one.

**Comment by rishabhmonga** [comment]
test this please

**Pull request merged**